src/main/java/com/amazon/opendistroforelasticsearch/sql/query/DefaultQueryAction.java

```diff
@@ -223,15 +235,45 @@ private void setSorts(List<Order> orderBys) {
 
                     request.addSort(
                             SortBuilders
-                                    .scriptSort(new Script(order.getName()), getScriptSortType(order))
-                                    .order(SortOrder.valueOf(order.getType())));
+                                    .scriptSort(new Script(orderByName), getScriptSortType(order))
+                                    .order(sortOrder));
                 } else {
-                    request.addSort(order.getName(), SortOrder.valueOf(order.getType()));
+                    if (orderByName.equals(ScoreSortBuilder.NAME)) {
+                        request.addSort(orderByName, sortOrder);
+                    } else {
+                        FieldSortBuilder fieldSortBuilder;
+                        if (sortBuilderMap.containsKey(orderByName)) {
+                            fieldSortBuilder = sortBuilderMap.get(orderByName);
+                            setSortParmas(fieldSortBuilder, order);
+                        } else {
+                            fieldSortBuilder = SortBuilders.fieldSort(orderByName);
+                            setSortParmas(fieldSortBuilder, order);
+                            sortBuilderMap.put(orderByName, fieldSortBuilder);
+                            request.addSort(fieldSortBuilder);
+                        }
+                    }
                 }
             }
         }
     }
 
+
+    private void setSortParmas(FieldSortBuilder fieldSortBuilder, Order order) {
+        fieldSortBuilder.order(SortOrder.valueOf(order.getType()));
+
+        SQLExpr expr = order.getSortField().getExpression();
+        if (expr instanceof SQLBinaryOpExpr) {
+            // we set SQLBinaryOpExpr in Field.setExpression() to support ORDER by IS NULL/IS NOT NULL
+            fieldSortBuilder.missing(getNullOrderString((SQLBinaryOpExpr) expr));
+        }
+
```

Comment:
inconsistency - that method has empty line before last } and next one doesn't.
Consider removing those empty lines here and everywhere